### PR TITLE
Ops 1088 edit agreement UI

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

Adds edit agreement functionality reusing CreateEditAgreement component.

## Issue

#1088 
( needed for #856 and #681 )

## How to test

1. Goto Agreements list
2. Select one by clicking on the pencil
3. See that data from agreement is now part of the agreement workflow
4. Edit data (in all three steps including the budge lines)
5. Save and see that the Agreement has been updated back in the Agreement list.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
         "vest": "4.6.11"
     },
     "devDependencies": {
+        "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
```
Comment (fpigeonjr) [inline frontend/package.json:34]: created #1111 to address this

File: frontend/src/components/UI/Form/ProcurementShopSelect/ProcurementShopSelect.jsx [hunk 2]
```diff
@@ -36,6 +81,7 @@ export const ProcurementShopSelect = ({ selectedProcurementShop, onChangeSelecte
                     id="procurement-shop-select"
                     onChange={handleChange}
                     value={selectedProcurementShop?.id}
+                    defaultValue={2}
```
Comment (fpigeonjr) [inline frontend/src/components/UI/Form/ProcurementShopSelect/ProcurementShopSelect.jsx:84]: defaults to GCS

File: frontend/cypress/e2e/editAgreement.cy.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            .then(cy.log);
+
+        cy.get("h1").should("have.text", "Edit Agreement");
+        cy.get("h2").first().should("have.text", "Budget Line Details");
+
+        cy.get('[data-cy="continue-btn"]').click();
+        cy.get("h1").should("have.text", "Agreement draft saved");
+        cy.get("h1").should("exist");
+
+        // TODO: DELETE test agreement (after API implemented)
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply frontend/cypress/e2e/editAgreement.cy.js:93]: 🤔I think it's dependent on #1036

[the remark this answers is not in the document]
Comment (stevtek) [reply frontend/cypress/e2e/editAgreement.cy.js:93]: right, it's a note for future work after #1036, then we can add in the DELETE here.  For that matter, maybe we should add DELETEs to the create-agreement tests, too.

[the remark this answers is not in the document]
Comment (fpigeonjr) [reply frontend/cypress/e2e/editAgreement.cy.js:93]: yes good idea

**Review by stevtek — APPROVED** [review]

**Pull request merged**